packages/edit-navigation/src/components/inspector-additions/auto-add-pages-panel.js

```diff
@@ -0,0 +1,37 @@
+/**
+ * WordPress dependencies
+ */
+import { useSelect, useDispatch } from '@wordpress/data';
+import { useState, useEffect } from '@wordpress/element';
+import { PanelBody, CheckboxControl } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+
+export default function AutoAddPagesPanel( { menuId } ) {
+	const menu = useSelect( ( select ) => select( 'core' ).getMenu( menuId ) );
+
+	const [ autoAddPages, setAutoAddPages ] = useState( null );
+
+	useEffect( () => {
+		if ( autoAddPages === null && menu ) {
+			setAutoAddPages( menu.auto_add );
+		}
+	}, [ autoAddPages, menu ] );
```

Comment:
The local state is so that the checkbox below can be a controlled component. We can't initialise this state to `menu.auto_add` because `menu` is `null` when the component mounts and when the REST API request is pending. We could remove the `useSelect()` by passing `menu` as a prop instead of `menuId` as a prop but the state would still have to remain because the `menu` prop is still potentially `null`.

Reply:
> We could remove the useSelect() by passing menu as a prop instead of menuId but it doesn't change much because the menu prop is still potentially null.

I suppose doing this would make the checkbox populate faster, though. OK I'll do that. 🙂

Reply:
Never mind me. I see what you mean now. Yes, we should be able to rely on `@wordpress/data` to persist this state.

Reply:
Hm. Annoyingly, when I do this, `getMenus()` starts a `GET /menus` request immediately after `saveMenu()` is dispatched. This clobbers the `auto_add` value that we just saved because the backend hasn't had a chance to persist the new value to the database.

I'll need to dig into why `@wordpress/data` does this and fix it there. It should be smart enough to not issue unnecessary API requests. I've made a note to come back to this later.